### DISCUSSION TITLE
Is it possible to use multiple Postgresql unnest functions together?

### DISCUSSION THREAD

**Discussion by soligto** [linq2db/linq2db#4360] [Q&A]
Hello!

I was trying to use multiple unnest functions together with linq2db but haven't figured out how to do so. Could anyone take a look please?

For example, I have two arrays: `string[] cities` and `string[] streets`. `streets[0]` corresponds to `cities[0]` and so on. Arrays have the same length.

I can bind these arrays and immediately convert them to rows using the following sql:

```sql
SELECT
  unnest({cities}::text[]) as city,
  unnest({streets}::text[]) as street
```

Linq2db does provide `unnest` [function](https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/DataProvider/PostgreSQL/PostgreSQLExtensions.cs#L111), but it seems to be impossible to express the above sql with this one.

Is there any other way?

**Comment by sdanyliv** [comment]
Actually it is simple to create such extension by yourself. 
Can you provide me with documentation of such behavior?
Found one example reasonable for me
```sql
SELECT * FROM
  unnest(
    ARRAY[1, 2, 3, 4],
    ARRAY['Shawn', 'Bible', 'Build', 'Jeff'],
    ARRAY[23, 24, 28, 27]
  ) AS data(id,name,age);
```
![image](https://github.com/linq2db/linq2db/assets/783759/2e0e736b-9dc4-49b6-926f-1d4086508f38)
